File: cpp/struktury.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


def osoba(imie, nazwisko, wiek):
    return{'imie': imie, 'nazwisko': nazwisko, 'wiek': wiek}


def pobierzDane(lista, n):
    for i in range(0, n):
        imie = input('Podaj imię:')
        nazwisko = input('Podaj nazwisko:')
        wiek = int(input('Podaj wiek:'))
        lista.append(osoba(imie, nazwisko, wiek))
    return lista


def wyswietlDane(lista):
    for o in lista:
        print("Imię: {:20}".format(o['imie']))
        print("Nazwisko: {:20}".format(o['nazwisko']))
        print("Wiek: {:20}".format(o['wiek']))
    print()


def zapiszDane(lista):
    with open('osobyp.txt', 'w') as plik:
        for o in lista:
            plik.write(
                o['imie'] + "," + o['nazwisko'] + "," + str(o['wiek']) + "\n")


def zapiszDaneCsv(lista):
    import csv
    with open('osobyp.csv', 'w', newline='') as plik:
        tresc = csv.writer(plik, delimiter=';')
        for o in lista:
            tresc.writerow(o.values())


def czytajDane(nazwa):
    import csv
    with open(nazwa, newline='') as plik:
        tresc = csv.reader(plik, delimiter=';')
        for o in tresc:
            print(o)


def main(args):
    # osoba1 = osoba('Jan', 'Kowalski', '23')
    # print(osoba1['imie'], osoba1['nazwisko'])
    lista = []
    ile = int(input('Ile osób wprowadzisz?'))
    lista = pobierzDane(lista, ile)
    wyswietlDane(lista)
    zapiszDaneCsv(lista)
    czytajDane('osobyp.csv')
    return 0


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/struktury.cpp
/*
 * struktury.cpp
 * 
 * Copyright 2018  <>
 */


#include <iostream>
#include <iomanip>
#include <cstdio>
#include <cmath>
#include <fstream>

using namespace std;

struct osoba {
    char imie[20];
    char nazwisko[20];
    int wiek;
};

void pobierzDane(osoba t[], int n) {
    for(int i = 0; i < n; i++) {
        cin.ignore(1);
        cout << "Podaj imię: ";
        cin.getline(t[i].imie, 20);
        cout << "Podaj nawisko: ";
        cin.getline(t[i].nazwisko, 20);
        cout << "Podaj wiek: ";
        cin >> t[i].wiek;
    }
}

void wyswietlDane(osoba t[], int n) {
    for(int i = 0; i < n; i++) {
        cout << "Imię: " << setw(20) << setfill('_') << t[i].imie << endl;
        cout << "Nazwisko: " << setw(18) << setfill('_') << t[i].nazwisko << endl;
        cout << "Wiek: " << setw(17) << setfill('_') << t[i].wiek << endl;
    }
}

osoba pobierzInf(osoba u) {
    cout << "Podaj imię: ";
    cin.getline(u.imie, 20);
    cout << "Podaj nawisko: ";
    cin.getline(u.nazwisko, 20);
    cout << "Podaj wiek: ";
    cin >> u.wiek;
    return u;
}

void wyswietlInf(osoba u) {
    cout << "Imię: " << setw(20) << setfill('_') << u.imie << endl;
    cout << "Nazwisko: " << setw(18) << setfill('_') << u.nazwisko << endl;
    cout << "Wiek: " << setw(17) << setfill('_') << u.wiek << endl;
}

void zapiszOsoby(osoba t[], int n) {
        ofstream plik("osoby.txt", ios::app);
        if (!plik.is_open()) {
            cout << "Błąd otwarcia pliku!";
        } else {
            for(int i = 0; i < n; i++){
                cout << t[i].imie << "," << t[i].nazwisko << "," << t[i].wiek << endl;
                plik << t[i].imie << "," << t[i].nazwisko << "," << t[i].wiek << endl;
            }
        }
}

void czytajDane(osoba t[]) {
    ifsteram plik ("osoby.txt");
    string 
}

int main(int argc, char **argv)
{
    int ile;
    cout << "Ile osób chcesz wprowadzić: "; cin >> ile;

    osoba uczniowie[ile];
    //uczen = pobierzInf(uczen);
    pobierzDane(uczniowie, ile);
    wyswietlDane(uczniowie, ile);
    zapiszOsoby(uczniowie, ile);
	return 0;
}

